yjit/src/codegen.rs

```diff
@@ -7847,6 +7875,7 @@ fn gen_invokesuper_specialized(
             gen_send_iseq(jit, asm, ocb, iseq, ci, frame_type, None, cme, Some(block), ci_flags, argc, None)
         }
         VM_METHOD_TYPE_CFUNC => {
+            // FIXME: why do we put unknown comptime class here?
```

Comment:
I think it's because it's not known. `known_recv_class` expects the receiver's class (otherwise things like `jit_rb_str_to_s` would be broken), but we aren't guarding against it on invokesuper (we only guard against a method entry in the local environment).

For this feature, we'd actually like to document `cme->defined_class` instead of the receiver's class, so we probably need to pass a different variable if we want to provide that information. Even though we don't pass `known_recv_class`, it's not safe to overload this variable for this purpose (because we don't guard it).